### PR TITLE
Shubha 🧙‍♀️

### PR DESCRIPTION
**Pull request by shubha-rajan**

### CODE REVIEW

File: lib/possible_bipartition.rb [hunk 1]
```diff
@@ -1,4 +1,43 @@
+# I looked at the leetcode solution in Python 
+# (https://leetcode.com/problems/possible-bipartition/solution/)
```
Comment (CheezItMan) [inline lib/possible_bipartition.rb:2]: Ok great, it's been a while since you saw graphs in class anyway.

File: lib/possible_bipartition.rb [hunk 2]
```diff
@@ -1,4 +1,43 @@
...
+        queue.push(node + 1)
+      end
+    end  
+    return true
 end
+
+
+def build_graph(dislikes)
+  nodes = Hash.new {|h,k| h[k] = [] } #had to look up how to do this on StackOverflow
+    dislikes.each do |dislike|
```
Comment (CheezItMan) [inline lib/possible_bipartition.rb]: Minor thing:  You don't need to indent here.

File: lib/possible_bipartition.rb [hunk 3]
```diff
@@ -1,4 +1,43 @@
...
+        if grouped[neighbor] == nil
+          grouped[neighbor] = !grouped[node]
+          queue.push(neighbor)
+        elsif grouped[neighbor] == grouped[node]
+          return false
+        end  
+      end     
+      if queue.empty? && node < graph.keys.max
+        queue.push(node + 1)
+      end
```
Comment (CheezItMan) [inline lib/possible_bipartition.rb]: This doesn't seem like the most efficient way to handle a disconnected graph.  Maybe also check to ensure that `grouped[node].nil?`

**Review by CheezItMan — COMMENTED** [review]
This definitely works and I like the edge-cases you have in your tests.  I had some minor quibbles and I was suspicious that it didn't handle all disconnected graphs, but it seems to do so.  Nice work!

File: lib/possible_bipartition.rb (hunk 3 above)
Comment (shubha-rajan) [reply lib/possible_bipartition.rb]: I updated this to seek the next ungrouped node. It gets rid of a lot of the aimless wandering of the last approach, but still doesn't seem very efficient. Thoughts?:
```
if queue.empty? 
    graph.keys.each do |key|
       if grouped[key].nil?
           queue.push(key)
           break
       end
    end
end
```
